tests/integration/targets/latest/tasks/object_metadata.yml

```diff
@@ -0,0 +1,197 @@
+          - test_three['object_metadata']['assigned_object_id'] == test_metadata_device['key']
+          - test_three['object_metadata']['scoped_fields'] == ["name", "serial"]
+          - test_three['msg'] == "object_metadata Foobar updated"
+
+    - name: "OBJECT METADATA 4: Update (Idempotent)"
+      networktocode.nautobot.object_metadata:
+        url: "{{ nautobot_url }}"
+        token: "{{ nautobot_token }}"
+        metadata_type: "TestMetadataTextType"
+        assigned_object_type: "dcim.device"
+        assigned_object_id: "{{ test_metadata_device['key'] }}"
+        value: "Foobar"
+        scoped_fields:
+            - name
+            - serial
+        state: present
+      register: test_four
+
+    - name: "OBJECT METADATA 4: ASSERT Update (Idempotent)"
+      assert:
+        that:
+          - not test_four['changed']
+          - test_four['object_metadata']['value'] == "Foobar"
+          - test_four['msg'] == "object_metadata Foobar already exists"
+
+    - name: "OBJECT METADATA 5: Test absent state"
+      networktocode.nautobot.object_metadata:
+        url: "{{ nautobot_url }}"
+        token: "{{ nautobot_token }}"
+        metadata_type: "TestMetadataTextType"
+        assigned_object_type: "dcim.device"
+        assigned_object_id: "{{ test_metadata_device['key'] }}"
+        value: "Foobar"
+        state: absent
+      register: test_five
+
+    - name: "OBJECT METADATA 5: ASSERT - Item removed"
+      assert:
+        that:
+          - test_five is changed
+          - test_five['diff']['before']['state'] == "present"
+          - test_five['diff']['after']['state'] == "absent"
+          - test_five['object_metadata']['metadata_type'] == test_metadata_type['key']
+          - test_five['object_metadata']['assigned_object_type'] == "dcim.device"
+          - test_five['object_metadata']['assigned_object_id'] == test_metadata_device['key']
+          - "'deleted' in test_five['msg']"
+
+    - name: "OBJECT METADATA 6: Test absent state (Idempotent)"
+      networktocode.nautobot.object_metadata:
+        url: "{{ nautobot_url }}"
+        token: "{{ nautobot_token }}"
+        metadata_type: "TestMetadataTextType"
+        assigned_object_type: "dcim.device"
+        assigned_object_id: "{{ test_metadata_device['key'] }}"
+        value: "Foobar"
+        state: absent
+      register: test_six
+
+    - name: "OBJECT METADATA 6: ASSERT - Item not removed (Idempotent)"
+      assert:
+        that:
+          - test_six is not changed
+          - "'already absent' in test_six['msg']"
+
+    - name: "OBJECT METADATA 7: Create contact type metadata"
+      networktocode.nautobot.object_metadata:
+        url: "{{ nautobot_url }}"
+        token: "{{ nautobot_token }}"
+        metadata_type: "TestMetadataContactType"
+        assigned_object_type: "dcim.device"
+        assigned_object_id: "{{ test_metadata_device['key'] }}"
+        contact: 
+          name: "My Contact"
+          phone: ""
+          email: ""
```

Comment:
Were these required?

Reply:
Yes, the unique set on that module requires name, phone, and email.

Reply:
That's odd, because I only use the name in the team tests here: https://github.com/nautobot/nautobot-ansible/blob/develop/tests/integration/targets/latest/tasks/team.yml#L80-L82. There was a bug up until v2.2.1 where they were set as required fields, but that has been fixed.

Reply:
I noticed that's `contacts` instead of `contact`.  Has it been fixed for both?

Reply:
Looks like it didn't work initially because I did not specify `name` like your example.  Should be good now.